lib/reverse_sentence.rb

```diff
@@ -1,4 +1,31 @@
 # A method to reverse the words in a sentence, in place.
 def reverse_sentence(my_sentence)
-  raise NotImplementedError
+
+  unless my_sentence.nil?
+    words = my_sentence.split(/(\s)/)
```

Comment:
The assignment instructions stated to not use any Ruby provided methods other than .length. Can you think of an approach without using .split?

The test cases with multiple, consecutive spaces in between words won't work with your algorithm since your algorithm does not preserve these spaces between words.

Also, .split is adding to the space complexity of your solution. How can you avoid the additional array of words along with split and join.